modules/extension/wmts/src/test/java/org/geotools/ows/wmts/online/WMTSOnlineTestCase.java

```diff
@@ -0,0 +1,89 @@
+package org.geotools.ows.wmts.online;
+
+import java.awt.Rectangle;
+import java.awt.image.RenderedImage;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.Set;
+import javax.media.jai.Interpolation;
+import org.geotools.coverage.grid.GridEnvelope2D;
+import org.geotools.coverage.grid.GridGeometry2D;
+import org.geotools.coverage.grid.io.AbstractGridFormat;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.ows.ServiceException;
+import org.geotools.ows.wms.CRSEnvelope;
+import org.geotools.ows.wmts.WebMapTileServer;
+import org.geotools.ows.wmts.map.WMTSCoverageReader;
+import org.geotools.ows.wmts.map.WMTSMapLayer;
+import org.geotools.ows.wmts.model.WMTSCapabilities;
+import org.geotools.ows.wmts.model.WMTSLayer;
+import org.geotools.ows.wmts.request.GetTileRequest;
+import org.geotools.parameter.Parameter;
+import org.geotools.test.OnlineTestCase;
+import org.geotools.tile.Tile;
+import org.opengis.parameter.GeneralParameterValue;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+public abstract class WMTSOnlineTestCase extends OnlineTestCase {
+
+    @Override
+    protected String getFixtureId() {
```

Comment:
This resembles how it originally was. I would like to keep the separate getFixtureId for different infrastructures. That will keep the tests running at other infrastructures if there is a failure within a single infrastructure. And I would also have a WMTSTestSupport class rather than a super class.

Reply:
Online tests in GeoTools commonly extend OnlineTestCase, please let's not add a new infrastructure for online test cases, many of us work across several modules, we cannot have each one roll their own. However, having a dedicated subclass for WMTS, with extra goodies specific for WMTS, may be a good idea.

Reply:
BTW, the method is protected and overridable, so different tests can have different fixture files, if needs be.
Ideally, I'd prefer to see something more self-contained, using containers (e.g., see ElasticSearch test, using testcontainers.org), rather that external servers whose contents can change completely out of our control... but I understand that's quite a bit more work, the WMTS server containers [would have to be prepared ad-hoc](https://www.testcontainers.org/features/creating_images/).